monai/apps/pathology/losses/hovernet_loss.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict
+
+import torch
+from torch.nn import CrossEntropyLoss
+from torch.nn import functional as F
+from torch.nn.modules.loss import _Loss
+
+from monai.losses import DiceLoss
+from monai.transforms import SobelGradients
+from monai.utils.enums import HoVerNetBranch
+
+
+class HoVerNetLoss(_Loss):
+    """
+    Loss function for HoVerNet pipeline, which is combination of losses across three branches
+
+    Args:
+
+    """
+
+    def __init__(
+        self,
+        lambda_hv_mse: float = 2.0,
+        lambda_hv_mse_grad: float = 1.0,
+        lambda_nuclei_ce: float = 1.0,
+        lambda_nuclei_dice: float = 1.0,
+        lambda_type_ce: float = 1.0,
+        lambda_type_dice: float = 1.0,
+    ) -> None:
+        self.lambda_hv_mse = lambda_hv_mse
+        self.lambda_hv_mse_grad = lambda_hv_mse_grad
+        self.lambda_nuclei_ce = lambda_nuclei_ce
+        self.lambda_nuclei_dice = lambda_nuclei_dice
+        self.lambda_type_ce = lambda_type_ce
+        self.lambda_type_dice = lambda_type_dice
+        super().__init__()
+
+        self.dice = DiceLoss(softmax=True, smooth_dr=1e-03, smooth_nr=1e-03, reduction="sum", batch=True)
+        self.ce = CrossEntropyLoss(reduction="mean")
+        self.sobel = SobelGradients(kernel_size=5)
+
+    def _compute_sobel(self, image: torch.Tensor) -> torch.Tensor:
```

Comment:
I would suggest to add doc-string for all the functions here.

Thanks.

Reply:
@Nic-Ma, we usually don't mandate docstring for private methods. Do you suggest to do it here?